src/modules/auth/auth.service.ts

```diff
@@ -27,7 +27,7 @@ export class AuthService {
   ) {}
 
   async signUp(@Body() AuthDto: AuthDto): Promise<User> {
-    const { password, email } = AuthDto;
+    const { password, email, role } = AuthDto;
```

Comment:
you will not know the role when you sign up, its next step

Reply:
I will know, because I changed the order, firstly you select a role and after sign up

Reply:
but its not according to requirements

Reply:
I fixed sign up order as required and make post request after choosing user role

Reply:
hmm I still confusing you will not have role after registration. example I register and close website without select role. what this will mean for you? I need register again?

Reply:
deleted